clients/client-python/tests/integration/test_gvfs_with_gcs_credential.py

```diff
@@ -0,0 +1,112 @@
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import logging
+import os
+from random import randint
+import unittest
+
+from gcsfs import GCSFileSystem
+
+from gravitino import Catalog, Fileset, GravitinoClient
+from gravitino.filesystem import gvfs
+from tests.integration.test_gvfs_with_gcs import TestGvfsWithGCS
+
+logger = logging.getLogger(__name__)
+
+
+def gcs_with_credential_is_configured():
+    return all(
+        [
+            os.environ.get("GCS_STS_SERVICE_ACCOUNT_JSON_PATH") is not None,
+            os.environ.get("GCS_STS_BUCKET_NAME") is not None,
+        ]
+    )
+
+
+@unittest.skipUnless(gcs_with_credential_is_configured(), "GCS is not configured.")
+class TestGvfsWithGCSCredential(TestGvfsWithGCS):
+    # Before running this test, please set the make sure gcp-bundle-x.jar has been
+    # copy to the $GRAVITINO_HOME/catalogs/hadoop/libs/ directory
+    key_file = os.environ.get("GCS_STS_SERVICE_ACCOUNT_JSON_PATH")
+    bucket_name = os.environ.get("GCS_STS_BUCKET_NAME")
+    metalake_name: str = "TestGvfsWithGCSCredential_metalake" + str(randint(1, 10000))
+
+    @classmethod
+    def _init_test_entities(cls):
+        cls.gravitino_admin_client.create_metalake(
+            name=cls.metalake_name, comment="", properties={}
+        )
+        cls.gravitino_client = GravitinoClient(
+            uri="http://localhost:8090", metalake_name=cls.metalake_name
+        )
+
+        cls.config = {}
+        cls.conf = {}
+        catalog = cls.gravitino_client.create_catalog(
+            name=cls.catalog_name,
+            catalog_type=Catalog.Type.FILESET,
+            provider=cls.catalog_provider,
+            comment="",
+            properties={
+                "filesystem-providers": "gcs",
+                "gcs-credential-file-path": cls.key_file,
+                "gcs-service-account-file": cls.key_file,
+                "credential-providers": "gcs-token",
+            },
+        )
+        catalog.as_schemas().create_schema(
+            schema_name=cls.schema_name, comment="", properties={}
+        )
+
+        cls.fileset_storage_location: str = (
+            f"gs://{cls.bucket_name}/{cls.catalog_name}/{cls.schema_name}/{cls.fileset_name}"
+        )
+        cls.fileset_gvfs_location = (
+            f"gvfs://fileset/{cls.catalog_name}/{cls.schema_name}/{cls.fileset_name}"
+        )
+        catalog.as_fileset_catalog().create_fileset(
+            ident=cls.fileset_ident,
+            fileset_type=Fileset.Type.MANAGED,
+            comment=cls.fileset_comment,
+            storage_location=cls.fileset_storage_location,
+            properties=cls.fileset_properties,
+        )
+
+        cls.fs = GCSFileSystem(token=cls.key_file)
+
+    def test_mkdir(self):
```

Comment:
why not move these tests to `TestGvfsWithHDFS` ?

Reply:
TestGvfsWithHDFS already has the method and [test_gvfs_with_gcs_credential.py](https://github.com/apache/gravitino/pull/5997/files/bd8618ec3c446ee3e9df4ce840fc2c4306b257a7#diff-c1af14e87a421259945e3e79d9c2f271743395b0a70eb1a957757d8028e29a19) just overwrite it.

Reply:
Could you provide the reason?

Reply:
GCS has different behaviour compared to hdfs, so we to overwrite it. OSS and S3 are similar.

Reply:
could you add detailed reason in the code about why adding the test in separate file.

Reply:
I can added it, not only `mkdir`, other operation like `makedirs`, `ls` also have similar issues.

Reply:
I have added some details on why we need to overwrite it.